### ISSUE TITLE
Mac M1 Docker problem

### ISSUE THREAD

**Issue by diananova** [chaoss/grimoirelab#481]
When running `sudo docker-compose up` on my Mac (with M1 chip), I get this error:

```
 runtime: failed to create new OS thread (have 2 already; errno=22)
```
I get this error only for Kibiter, everything else seems to be working. 
<img width="978" alt="Screenshot 2022-03-26 at 20 52 49" src="https://user-images.githubusercontent.com/24480679/160255044-056b8cf6-e703-4173-bf82-dcab2754484c.png">

**Comment by vchrombie** [comment]
Hi @diananova, the GrimoireLab is built and tested on Linux-based systems. We cannot give much help since we cannot reproduce the problem in our local system.

It would also be helpful if you can post the logs (if available).

Quick google search around this topic gave me some results, which suggests to tweak the Dockerfile
- https://stackoverflow.com/questions/66349391/failed-to-create-new-os-thread-have-2-already-errno-22
- https://serveanswer.com/questions/runtime-failed-to-create-new-os-thread-have-2-already-errno-22-in-docker-elasticsearch
- https://stackoverflow.com/questions/70093106/runtime-failed-to-create-new-os-thread-have-2-already-errno-22-in-docker-el

It would be great if you can try the solutions and let us know what worked.
Please let us know if you need any more help from us.

**Comment by vchrombie** [comment]
Cross posting this message from Slack

> Sean: Make sure you are using Docker for Apple Silicon: https://docs.docker.com/desktop/mac/apple-silicon/

@diananova feel free to try this and please do let us know if there is any progress.

**Comment by vchrombie** [comment]
Hi @sharontao0420 
Georg, one of our community members, has written a post on how to [install GrimoireLab using docker-compose](https://chaoss.community/blog-post/2019/10/08/simplifying-how-to-install-grimoirelab/). He uses a Mac but I'm not sure he uses Mac with N M1 chip (prolly not).

We are a bit helpless because none of the GrimoireLab maintainers use Mac and it is really hard for us to reproduce the problem and try finding a solution. Moreover, I understand that this is the problem with docker and Mac, but GrimoireLab still works fine for the rest of the systems. I really don't know if we can help with this problem in any way. Thanks for understanding.

I would still keep this issue open, with a hope there would a solution to this.

**Comment by diananova** [comment]
Yeah using Docker for Apple Silicon doesn't solve the issue.

**Comment by hogepodge** [comment]
I'm guessing that the Kibiter docker image needs to be rebuilt for ARM. Digging into the Dockerfile, it uses an x86 Kibiter build.

**Comment by loganknecht** [comment]
Hello!

I would like to communicate that this is an issue for me as well. This is really unfortunate because this project seems very exciting!

**Comment by loganknecht** [comment]
For anyone who needs to use an Apple M1 Docker setup, please try using these files to use `opensearch`

- https://github.com/chaoss/grimoirelab/blob/master/docker-compose/docker-compose-opensearch.yml
- https://github.com/chaoss/grimoirelab/blob/master/default-grimoirelab-settings/setup-opensearch.cfg

You won't have the default dashboards though when using open search. In order to that you can follow the steps here
- https://github.com/chaoss/grimoirelab/issues/547#issuecomment-1344958661

**Comment by sduenas** [comment]
Images should work now for M1. Feel free to open this ticket again if you find the same problem.

**Comment by mhauru** [comment]
When running `docker compose up` on my ARM Mac, the elasticsearch container fails with
```
elasticsearch-1      | [2024-04-25T10:37:39,214][INFO ][o.e.t.TransportService   ] [k_2zKtx] publish_address {172.21.0.2:9300}, bound_addresses {[::]:9300}
elasticsearch-1      | [2024-04-25T10:37:39,226][INFO ][o.e.b.BootstrapChecks    ] [k_2zKtx] bound or publishing to a non-loopback address, enforcing bootstrap checks
elasticsearch-1      | ERROR: [1] bootstrap checks failed
elasticsearch-1      | [1]: system call filters failed to install; check the logs and fix your configuration or disable system call filters at your own risk
elasticsearch-1      | [2024-04-25T10:37:39,233][INFO ][o.e.n.Node               ] [k_2zKtx] stopping ...
elasticsearch-1      | [2024-04-25T10:37:39,373][INFO ][o.e.n.Node               ] [k_2zKtx] stopped
elasticsearch-1      | [2024-04-25T10:37:39,373][INFO ][o.e.n.Node               ] [k_2zKtx] closing ...
elasticsearch-1      | [2024-04-25T10:37:39,391][INFO ][o.e.n.Node               ] [k_2zKtx] closed
elasticsearch-1 exited with code 78
```

I don't really know what's going on there, but I suspect this is related to the fact that the elasticsearch 6.8.6 docker container is only available for amd64 and on arm gets run through Docker's emulation thingie. (Even if that's not the cause of the fail, the emulation is sure to a significant cause a performance hit.)

Elasticsearch v7 images are available on Docker Hub for arm64 as well, so upgrading to v7 would solve this. Don't know how disruptive that would be for GrimoireLab.

I haven't tried the opensearch containers as a workaround. If that's the recommended way to run on ARM, that would be good to have in the documentation.

**Comment by sduenas** [comment]
Yes, ElasticSearch docker images don't have images for ARM. You can use OpenSearch or ElasticSearch 7. However, you will have to upload the dashboards manually, which is not great. I still think you should use them, though.

**Comment by mhauru** [comment]
Any plans to upgrade to ElasticSearch 7, or 8, in general?

If I find a time I might submit a PR for docs to mention this for ARM users.

**Comment by SamuZad** [comment]
If anyone comes across this issue in the future.. Rather than using the mac-specific docker compose file (which does not include dashboards, per the documentation), follow these steps:
1. in your terminal going to run docker compose, set  `export DOCKER_DEFAULT_PLATFORM=linux/amd64`
2. in the docker-compose file, make sure to add `-Ebootstrap.system_call_filter=false `to the command, such that:
```
    elasticsearch:
      image: docker.elastic.co/elasticsearch/elasticsearch-oss:6.8.6
      command: elasticsearch -Enetwork.bind_host=0.0.0.0 -Ehttp.max_content_length=2000mb -Ebootstrap.system_call_filter=false
      ports:
        - 9200:9200
      environment:
        - ES_JAVA_OPTS=-Xms2g -Xmx2g
        - ANONYMOUS_USER=true
```
3. run docker-compose up 🙂 

This way you will have an instance with dashboards initialized by default on a Mac!
